File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPrestamosBO/PenalizacionBO.cs
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftBibliotecaPrestamosBO
{
    public class PenalizacionBO : BaseBO
    {
        /*
        public int insertar(string descripcion, float monto, int idPrestamoAsociado)
        {
            return this.WsCliente.penalizacion_insertar(descripcion, monto, idPrestamoAsociado);
        }

        public int modificar(int idPenalizacion, string descripcion, float monto, int idPrestamoAsociado)
        {
            return this.WsCliente.penalizacion_modificar(idPenalizacion, descripcion, monto, idPrestamoAsociado);
        }

        public int eliminar(int idPenalizacion)
        {
            return this.WsCliente.penalizacion_eliminar(idPenalizacion);
        }

        public BindingList<penalizacion> listarTodos()
        {
            try
            {
                penalizacion[] penalizaciones = this.WsCliente.penalizacion_listarTodos();
                if (penalizaciones == null) return new BindingList<penalizacion>();
                return new BindingList<penalizacion>(penalizaciones);
            }
            catch (Exception ex)
            {
                return new BindingList<penalizacion>();
            }
        }

        public BindingList<penalizacion> listarTodos(int idPrestamoAsociado)
        {
            try
            {
                penalizacion[] penalizaciones = this.WsCliente.penalizacion_listarTodos_porParametros(idPrestamoAsociado);
                if (penalizaciones == null) return new BindingList<penalizacion>();
                return new BindingList<penalizacion>(penalizaciones);
            }
            catch (Exception ex)
            {
                return new BindingList<penalizacion>();
            }
        }

        public penalizacion obtenerPorId(int idPenalizacion)
        {
            return this.WsCliente.penalizacion_obtenerPorId(idPenalizacion);
        }
        */

        public BindingList<penalizacion> verPenalizacionesPrestamo(int idPrestamoAsociado)
        {
            try
            {
                penalizacion[] penalizaciones = this.WsCliente.verPenalizacionesPrestamo(idPrestamoAsociado);
                if (penalizaciones == null) return new BindingList<penalizacion>();
                return new BindingList<penalizacion>(penalizaciones);
            }
            catch (Exception ex)
            {
                return new BindingList<penalizacion>();
            }
        }
    }
}

File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/PenalizacionBO.java
package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.prestamos.model.Penalizacion;
import java.rmi.Remote;
import java.rmi.RemoteException;

public interface PenalizacionBO extends Remote{
    
    public Integer insertar(String descripcion,Float monto,Integer idPrestamoAsociado,String fechaImposicion, Integer estaActivo) throws RemoteException;
    
    public Integer modificar(Integer idPenalizacion,String descripcion,Float monto,Integer idPrestamoAsociado, String fechaImposicion, Integer estaActivo) throws RemoteException;
    
    public Integer eliminar(Integer idPenalizacion) throws RemoteException;
    
    public ArrayList<Penalizacion> listarTodos(Integer idPrestamoAsociado,String fechaImposicion, Integer estaActivo, Integer limite) throws RemoteException;
    
    public Penalizacion obtenerPorId(Integer idPenalizacion) throws RemoteException;
    
    public ArrayList<Penalizacion> verPenalizacionesPrestamo(Integer idPrestamoAsociado) throws RemoteException;
    
}
